lib/systems/default.nix

```diff
@@ -76,6 +77,44 @@ let
     # TODO: deprecate args.rustc in favour of args.rust after 23.05 is EOL.
     rust = args.rust or args.rustc or {};
 
+    toolchain = {
+      cc = if final.useLLVM || final.isDarwin then "clang"
+        else if final.useArocc then "arocc"
+        else if final.useZig then "zig"
+        else "gcc";
+
+      bintools = if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then
+        "llvm"
+      else "gnu";
+
+      cxxlib =
+        /**/ if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then "libcxx"
+        else "libstdcxx";
+
+      unwinderlib =
+        /**/ if final.useLLVM || final.useArocc || final.useZig then "libunwind"
+        else if final.isDarwin then "libunwind-system"
+        else "libgcc_s";
```

Comment:
I think this and maybe other things might be wrong for MinGW + threads + etc., but I’m not sure what would be right and it can probably wait until this work actually starts touching bootstrap in a way that will flag it up.

Reply:
Yeah, I think @Ericson2314 would know? However, he can be busy and difficult to get ahold of sometimes. If he can clarify like before the end of the week then I think we can go from there but if it's longer than a week then follow up PR's can fix it.